src/argus_htmx/auth/views.py

```diff
@@ -0,0 +1,50 @@
+from django.conf import settings
+from django.contrib.auth.views import LoginView as DjangoLoginView
+from django.urls import reverse
+
+from argus.auth.utils import (
+    has_model_backend,
+    has_remote_user_backend,
+    get_psa_authentication_backends,
+    get_authentication_backend_classes,
+)
+
+
+REMOTE_USER_METHOD_NAME = getattr(settings, "ARGUS_REMOTE_USER_METHOD_NAME", "REMOTE_USER")
+
+
+def get_htmx_authentication_backend_name_and_type():
+    # Needed for HTMX LoginView
+    backends = get_authentication_backend_classes()
+
+    data = {}
+    if has_model_backend(backends):
+        data["username_password"] = {
+            "url": reverse("htmx:login"),
+            "button": "Log In",
+        }
+
+    if has_remote_user_backend(backends):
+        remote_user_data = {
+            "url": "/",
+            "button": REMOTE_USER_METHOD_NAME,
+        }
+        data.setdefault("link", []).append(remote_user_data)
+
+    for backend in get_psa_authentication_backends(backends):
+        psa_backend_data = {
+            "url": reverse("social:begin", kwargs={"backend": backend.name}),
+            "button": backend.name,
+        }
+        data.setdefault("link", []).append(psa_backend_data)
+
+    return data
```

Comment:
The structure of `data` seems a bit weird. If I read it correctly it will take the form of

```python
{
  "username_password": {"url": "/some/url", "button": "Log In",
  "link": [
    {
      "url": "/some/other/url",
      "button": "other auth provider"
    }
}
```
The current naming puts a backend name "username_password", on the same level as "link". When I read link, I expect a single link/url, but instead I get a list of other backends...

maybe some renames would fix it? `username_password` -> `local`, `link` -> `other` or `additional`

Also, the "button" is not a button, but text. Should it be called `name` or `display_name`?

Reply:
Good suggestions!

Reply:
I went with `local`, `external`, and `display_name`.